cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/validation/ValidationFailure.java

```diff
@@ -73,12 +76,28 @@ public ValidationFailure withCause(Cause cause) {
    * @param pluginId plugin id
    * @param pluginName plugin name
    * @param pluginType plugin type
+   * @param requestedArtifact requested artifact
+   * @param suggestedArtifact suggested artifact
    * @return validation failure with plugin not found cause
    */
-  public ValidationFailure withPluginNotFound(String pluginId, String pluginName, String pluginType) {
-    causes.add(new Cause().addAttribute(CauseAttributes.PLUGIN_ID, pluginId)
-                 .addAttribute(CauseAttributes.PLUGIN_NAME, pluginName)
-                 .addAttribute(CauseAttributes.PLUGIN_TYPE, pluginType));
+  public ValidationFailure withPluginNotFound(String pluginId, String pluginName, String pluginType,
```

Comment:
probably worth having a convenience method without the requested and suggested artifact arguments.